### PR TITLE
Add LegalHold API Support

### PR DESCRIPTION
**Pull request by nitisht**
Add Legal Hold API Support

### CODE REVIEW

File: examples/s3/putobjectlegalhold.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// +build ignore
+
+/*
+ * MinIO Go Library for Amazon S3 Compatible Cloud Storage
+ * Copyright 2019 MinIO, Inc.
```
Comment (kannappanr) [inline examples/s3/putobjectlegalhold.go]:
```suggestion
 * Copyright 2020 MinIO, Inc.
```

File: go.sum [hunk 2]
```diff
@@ -7,6 +7,7 @@ github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1/go.mod h1:wJfORR
 github.com/jtolds/gls v4.20.0+incompatible h1:xdiiI2gbIgH/gLH7ADydsJ1uDOEzR8yvV7C0MuV77Wo=
 github.com/jtolds/gls v4.20.0+incompatible/go.mod h1:QJZ7F/aHp+rZTRtaJ1ow/lLfFfVYBRgL+9YlvaHOwJU=
 github.com/konsorten/go-windows-terminal-sequences v1.0.1/go.mod h1:T0+1ngSBFLxvqU3pZ+m/2kptfBszLMUkC4ZK/EgS/cQ=
+github.com/minio/minio-go v6.0.14+incompatible h1:fnV+GD28LeqdN6vT2XdGKW8Qe/IfjJDswNVuni6km9o=
```
Comment (kannappanr) [inline go.sum]: is this required?

File: examples/s3/getobjectlegalhold.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// +build ignore
+
+/*
+ * MinIO Go Library for Amazon S3 Compatible Cloud Storage
+ * Copyright 2019 MinIO, Inc.
```
Comment (kannappanr) [inline examples/s3/getobjectlegalhold.go]:
```suggestion
 * Copyright 2020 MinIO, Inc.
```

File: go.sum (hunk 2 above)
Comment (nitisht) [reply go.sum]: This was inadvertent, let me remove this

Comment (nitisht) [reply go.sum]: done

File: docs/API.md [hunk 4]
```diff
@@ -1279,6 +1281,57 @@ if err != nil {
...
+__Parameters__
+
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`bucketName`  | _string_  |Name of the bucket  |
+|`objectName` | _string_  |Name of the object |
+|`opts`	|_minio.PutObjectLegalHoldOptions_ |Allows user to set options like status and version id |
+
+__minio.PutObjectLegalHoldOptions_
```
Comment (poornas) [inline docs/API.md]:
```suggestion
_minio.PutObjectLegalHoldOptions_
```

File: docs/API.md [hunk 5]
```diff
@@ -1279,6 +1281,57 @@ if err != nil {
...
+
+|Field | Type | Description |
+|:--- |:--- | :--- |
+| `opts.Status` | _*minio.LegalHoldStatus_ |Legal-Hold status to be set|
+| `opts.VersionID` | _string_ |Version ID of the object to apply retention on|
+
+```go
+s := minio.LegalHoldEnabled
+opts := minio.PutObjectRetentionOptions {
```
Comment (poornas) [inline docs/API.md]:
```suggestion
opts := minio.PutObjectLegalHoldOptions {
```

**Pull request merged**